### PR TITLE
Accept k8s version to build node images

### PR DESCRIPTION
**Pull request by kishen-v**
Changes: 
1. Accept version of k8s as an input to the workflow to build kind-node images.
2. Refer to `vars.QUAY_USERNAME` from the earlier `secrets.QUAY_USERNAME` (requires changes in the repository setting to have this variable defined)

The docker workflow tends to fail as k8s-version is defaulting to `null` rather than `master`. However, subsequent builds seem to run fine.

```
##[debug]Evaluating: format('Build kind-node image - {0}', inputs.k8s-version)
[50](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:54)
##[debug]Evaluating format:
[51](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:55)
##[debug]..Evaluating String:
[52](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:56)
##[debug]..=> 'Build kind-node image - {0}'
[53](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:57)
##[debug]..Evaluating Index:
[54](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:58)
##[debug]....Evaluating inputs:
[55](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:59)
##[debug]....=> Object
[56](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:60)
##[debug]....Evaluating String:
[57](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:61)
##[debug]....=> 'k8s-version'
[58](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:62)
##[debug]..=> null.   <--------------------
[59](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:63)
##[debug]=> 'Build kind-node image - '
[60](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:64)
##[debug]Result: 'Build kind-node image - '
[61](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:65)
##[debug]Set step '__run_3' display name to: 'Build kind-node image - '
[62](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:66)
##[debug]Set step '__run_4' display name to: 'Publish node image'
[63](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:67)
Complete job name: build
[64](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:68)
##[debug]Collect running processes for tracking orphan processes.
[65](https://github.com/kishen-v/kind-image/actions/runs/9937700804/job/27450385411#step:1:69)
##[debug]Finishing: Set up job
```
vs.
```
##[debug]Evaluating format:
[51](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:55)
##[debug]..Evaluating String:
[52](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:56)
##[debug]..=> 'Build kind-node image - {0}'
[53](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:57)
##[debug]..Evaluating Index:
[54](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:58)
##[debug]....Evaluating inputs:
[55](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:59)
##[debug]....=> Object
[56](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:60)
##[debug]....Evaluating String:
[57](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:61)
##[debug]....=> 'k8s-version'
[58](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:62)
##[debug]..=> 'master' <------------
[59](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:63)
##[debug]=> 'Build kind-node image - master'
[60](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:64)
##[debug]Result: 'Build kind-node image - master'
[61](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:65)
##[debug]Set step '__run_3' display name to: 'Build kind-node image - master'
[62](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:66)
##[debug]Set step '__run_4' display name to: 'Publish node image'
[63](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:67)
Complete job name: build
[64](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:68)
##[debug]Collect running processes for tracking orphan processes.
[65](https://github.com/kishen-v/kind-image/actions/runs/9938312957/job/27450516466#step:1:69)
##[debug]Finishing: Set up job
```

image

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
         uses: docker/login-action@v3
         with:
           registry: ${{ env.REGISTRY }}
-          username: ${{ secrets.QUAY_USERNAME }}
+          username: ${{ vars.QUAY_USERNAME }}
```
Comment (mkumatag) [inline .github/workflows/docker-publish.yml:49]: where is this variable getting added?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -69,13 +69,13 @@ jobs:
...
         run: |
           mkdir -p tmp/kubernetes
-          git clone --single-branch --filter=tree:0 --branch ${{ matrix.version }} https://github.com/kubernetes/kubernetes tmp/kubernetes
-          kind build node-image $PWD/tmp/kubernetes --arch ppc64le --image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ matrix.version }} --base-image ${{ env.BASE_IMAGE }}
+          git clone --single-branch --filter=tree:0 --branch ${{ inputs.k8s-version }} https://github.com/kubernetes/kubernetes tmp/kubernetes
+          kind build node-image $PWD/tmp/kubernetes --arch ppc64le --image ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ inputs.k8s-version }} --base-image ${{ env.BASE_IMAGE }}
 
       - name: Publish node image
-        if: github.event_name != 'pull_request'
+        if: github.event_name != 'pull_request' || github.event_name != 'push'
```
Comment (mkumatag) [inline .github/workflows/docker-publish.yml:79]: can we add comment why are we not pushing the image for any commit?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (kishen-v) [reply .github/workflows/docker-publish.yml:49]: Hi @mkumatag,
We may need to add this variable in repository settings to have the same defined.

File: .github/workflows/docker-publish.yml (hunk 2 above)
Comment (kishen-v) [reply .github/workflows/docker-publish.yml:79]: Done!

**Comment by mkumatag** [comment]
@kishen-v I made as an admin to this repo, you can add the required variables in the settings section and also build is failing in the CI, please check..

File: .github/workflows/docker-publish.yml [hunk 3]
```diff
@@ -69,13 +73,15 @@ jobs:
...
 
-      - name: Publish node image
-        if: github.event_name != 'pull_request'
+      - name: Publish node image - ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.K8S-VERSION }}
+        # Modifying or adding new changes to the workflow does not necessarily require the built image to be pushed.
+        # Images can solely be built through workflow dispatch.
+        if: github.event_name != 'pull_request' && github.event_name != 'push'
         run: |
-          docker push ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ matrix.version }}
+          docker push ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.K8S-VERSION }}
```
Comment (mkumatag) [inline .github/workflows/docker-publish.yml:87]: please add a new line in the end

**Review by mkumatag — CHANGES_REQUESTED** [review]
hope this PR is ready for the merge!

File: .github/workflows/docker-publish.yml (hunk 3 above)
Comment (kishen-v) [reply .github/workflows/docker-publish.yml]: Addressed, thanks!

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**